app/orchestrators/CorrectReturnOrchestrator.scala

```diff
@@ -20,18 +20,76 @@ import cats.data.EitherT
+import models.enums.SiteTypes.{PRODUCTION_SITE, WAREHOUSE}
+import models.requests.DataRequest
+import models.submission.VariationsSites.getHighestRefNumber
+import models.submission.{ClosedSite, ReturnVariationData, VariationsPersonalDetails, VariationsSite, VariationsSites, VariationsSubmission}
+import models.updateRegisteredDetails.ContactDetails
 import models.{ReturnPeriod, SdilReturn, UserAnswers}
+import pages.correctReturn.{CorrectionReasonPage, RepaymentMethodPage}
+import play.api.mvc.AnyContent
+import play.api.mvc.Results.InternalServerError
 import service.VariationResult
 import services.SessionService
 import uk.gov.hmrc.http.HeaderCarrier
+import utilities.GenericLogger
 
+import java.time.Instant
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class CorrectReturnOrchestrator @Inject()(connector: SoftDrinksIndustryLevyConnector,
-                                          sessionService: SessionService){
+                                          sessionService: SessionService,
+                                          genericLogger: GenericLogger,
+                                          val errorHandler: ErrorHandler){
+
+  def submitVariation()
+                     (implicit request: DataRequest[AnyContent], hc: HeaderCarrier, ec: ExecutionContext): Future[Unit] = {
+    val returnVariationData = for {
+      originalReturn <- request.userAnswers.getCorrectReturnOriginalSDILReturnData
+      returnPeriod <- request.userAnswers.correctReturnPeriod
+      revisedReturn <- request.userAnswers.getCorrectReturnData
+    } yield getReturnsVariationToBeSubmitted(
+      subscription = request.subscription,
+      userAnswers = request.userAnswers,
+      originalReturn = originalReturn,
+      returnPeriod = returnPeriod,
+      revisedReturn = SdilReturn(
+        ownBrand = revisedReturn.howManyOperatePackagingSiteOwnBrands.map(litres => (litres.lowBand, litres.highBand)).getOrElse(0, 0),
+        packLarge = revisedReturn.howManyPackagedAsContractPacker.map(litres => (litres.lowBand, litres.highBand)).getOrElse(0, 0),
+        packSmall = request.userAnswers.smallProducerList,
+        importLarge = revisedReturn.howManyCreditsForLostDamaged.map(litres => (litres.lowBand, litres.highBand)).getOrElse(0, 0),
+        importSmall = revisedReturn.howManyBroughtIntoUkFromSmallProducers.map(litres => (litres.lowBand, litres.highBand)).getOrElse(0, 0),
+        export = revisedReturn.howManyClaimCreditsForExports.map(litres => (litres.lowBand, litres.highBand)).getOrElse(0, 0),
+        wastage = revisedReturn.howManyCreditsForLostDamaged.map(litres => (litres.lowBand, litres.highBand)).getOrElse(0, 0),
+        submittedOn = Some(Instant.now())
+      )
+    )
+    returnVariationData match {
+      case Some(returnVariation) =>  Future.successful(connector.submitReturnsVariation(request.subscription.sdilRef,  returnVariation))
+      case None => genericLogger.logger.info(s"failed to collect userAnswers")
```

Comment:
Perhaps add some details about the function where this logging takes place, similar to in the `submitReturnsVariation` for example